src/Roew/WrappedResult.php

```diff
@@ -57,7 +57,7 @@ public function getResult()
      *
      * @throws \LogicException
      */
-    public function getException(): \Exception
+    public function getException(): void
```

Comment:
It's strange. This method will throw the `\Exception`, and this doesn't return anything.

The type hint is `Exception` is not proper I think.

Reply:
It is. according to `ResultOrExceptionWrapper`, this method should return an exception in case the wrapper is a failure ( `WrappedException` ), and throw an exception in case its a success ( `WrappedResult` )

Reply:
I think we can use `?\Exception` is better than original approach.

Reply:
`?\Exception` says that `getException()` "might" return null, but that's never the cause, it always return an `\Exception`, or throws. that's why `isSuccessful()` and `isFailure()` exists :)

i would keep it the way it is.

Reply:
Agree. Revert this.